content/en/docs/architecture/images.md

```diff
@@ -0,0 +1,124 @@
+    upstream_image: registry.ci.openshift.org/ocp/builder:rhel-8-golang-openshift-{MAJOR}.{MINOR}
+
+  rhel:
+    image: openshift/ose-base:rhel8.2.els.rhel
+    upstream_image: registry.ci.openshift.org/ocp/builder:rhel-8-base-openshift-{MAJOR}.{MINOR}
+  ```
+
+  The ART team mirrors all images, such as `golang`, `golang-1.14`, and `rhel` defined in `streams.yml`, to `registry.ci.openshift.org`. The `upstream_image` field defines the reference of the mirrored image. Those images are colloquially known as _the ART equivalent images_ to ensure CI images are built from the exact same images as productized images.
+
+## Ensuring Identical Image Builds
+While the CI and productized image builds are configured in two separate places, a number of automatic processes exist to ensure that the container images built in both environments are identical. Automation enforces that the same build manifest (Dockerfile) is used to define the image in both systems. Furthermore, automation also ensures that the builder images and base images that are used in CI are identical to those used by ART when building productized container images in OSBS. Therefore, with identical container build manifests and input container images, identical builds run in CI and the productized environment.
+
+Moreover, a developer can build the same container image locally as well if the image build tool such as `docker` or `podman` has the [permission of pulling the images](/docs/how-tos/use-registries-in-build-farm/#how-do-i-log-in-to-pull-images-that-require-authentication) used in the build manifest. The images in `ocp` namespace are open to all authenticated users.
+
+## Determining Go Versions
+In the above example, the build of `cluster-etcd-operator` uses Go 1.15 to build the image, as configured in `streams.yml`. When the version of Go used to build OCP changes. the ART team will modify the version configured in `ocp-build-data` centrally for the whole product at once. Then, automation will create pull requests to update `Dockerfile`s in component repositories and their CI configuration:
+
+* a PR to the component repository modifies the `FROM` directive in `Dockerfile`. CI jobs on this pull request will reflect the use of the new Go version and repository administrators are expected to monitor the change, merging it if appropriate or landing changes as necessary to merge the change themselves. After code freeze in the release, any un-updated `FROM` entries become blocker bugs for the release. If an OCP component cannot migrate to the new toolchain version, the owner has to get the approval from the architecture team and then communicate with the ART team. In this case, the ART team will modify the builder image of the component in `ocp-build-data` and the created PR to the component repository can be closed. For example, `cluster-etcd-operator` has to use `golang-1.14`. `images/cluster-etcd-operator.yml` will has the following stanza:
```

Comment:
These are pretty long and should just be paragraphs, not bullets. Something like:

```
Then, automation will create pull requests to update Dockerfiles in component repositories and their CI configuration.

First, a pull request is filed to the component repository, where it modifies the FROM directive...
```